input/kafkamdm/kafkamdm.go

```diff
@@ -221,6 +235,14 @@ func (k *KafkaMdm) consumePartition(topic string, partition int32, partitionOffs
 			if err := offsetMgr.Commit(topic, partition, currentOffset); err != nil {
 				log.Error(3, "kafka-mdm failed to commit offset for %s:%d, %s", topic, partition, err)
 			}
+			partitionOffset[partition].Set(int(currentOffset))
```

Comment:
you cant use a slice as in a clustered environment a node would only be consuming from a subset of partitions.